### ISSUE TITLE
code = compile(evaluateFalse(code), '<string>', 'eval') is not working

### ISSUE THREAD

**Issue by proy87** [sympy/sympy#20594]
`code = compile(evaluateFalse(code), '<string>', 'eval')` from parsing module is not working under python 3.8

**Comment by oscarbenjamin** [comment]
What is `code`?

**Comment by proy87** [comment]
It is from `parse_expr`: `from sympy.parsing.sympy_parser import parse_expr`

**Comment by oscarbenjamin** [comment]
Can you show a complete example of code that demonstrates the problem including all imports and with all variables defined?

**Comment by proy87** [comment]
`parse_expr('x', evaluate=False)`

**Comment by oscarbenjamin** [comment]
I just tried that under Python 3.8 and it seemed to work fine:
```python
$ python
Python 3.8.1 (v3.8.1:1b293b6006, Dec 18 2019, 14:08:53) 
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from sympy import parse_expr
>>> parse_expr('x', evaluate=False)
x
```
Does that not work for you?

If you meant something else can you demonstrate what you mean unambiguously by showing exactly what you imported, what you defined and what functions you called (as I did above)?

**Comment by proy87** [comment]
I have Python 3.8.5 and the error is similar to this: https://github.com/nestorsalceda/mamba/issues/150

**Comment by proy87** [comment]
`ValueError: Name node can't be used with 'False' constant`

**Comment by oscarbenjamin** [comment]
I don't understand how to reproduce the problem based on your description. Can you make a small .py file that demonstrates the issue if I run it as `python t.py` and then show here what the contents of that file are?

**Comment by proy87** [comment]
```
>>> parse_expr('1+I', evaluate=False)
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/user/venv/lib/python3.8/site-packages/sympy/parsing/sympy_parser.py", line 1006, in parse_expr
    code = compile(evaluateFalse(code), '<string>', 'eval')
ValueError: Name node can't be used with 'False' constant
```

**Comment by oscarbenjamin** [comment]
I can't reproduce that. It works for me with Python 3.8.1 and sympy 1.7.1 on OSX:
```
In [1]: parse_expr('1+I', evaluate=False)                                                                                         
Out[1]: 1 + ⅈ
```

**Comment by proy87** [comment]
With Python 3.8.5 and sympy 1.6 it doesn't work.

**Comment by oscarbenjamin** [comment]
Okay I can reproduce this. It looks like it was fixed in 1.6.1 by 0807b36b93e57521007a69e25b616efa5b5ad7d2 https://github.com/sympy/sympy/pull/18819

I think if you upgrade sympy the issue is already fixed. The latest version of sympy is 1.7.1 which was released yesterday and your example works fine there:
```
In [1]: parse_expr('1+I', evaluate=False)
Out[1]: 1 + ⅈ
```

**Comment by proy87** [comment]
Great!. Thank you.
